File: pizza/pizza.go
package pizza

type Pizza struct {
	Name        string       `json:"name"`
	Dough       Dough        `json:"dough"`
	Ingredients []Ingredient `json:"ingredients"`
	Tool        string       `json:"tool"`
}

type Dough struct {
	Name             string `json:"name"`
	CaloriesPerSlice int    `json:"caloriesPerSlice"`
}

type Ingredient struct {
	Name             string `json:"name"`
	CaloriesPerSlice int    `json:"caloriesPerSlice"`
	Vegetarian       bool   `json:"vegetarian"`
}

func (p Pizza) IsVegetarian() bool {
	for _, ingredient := range p.Ingredients {
		if !ingredient.Vegetarian {
			return false
		}
	}
	return true
}

func (p Pizza) CalculateCalories() int {
	calories := 0
	for _, ingredient := range p.Ingredients {
		calories += ingredient.CaloriesPerSlice
	}
	return calories
}

File: Mobiles/flutter/lib/models/pizza.dart
class Pizza {
  final int id;
  final String name;
  final Dough dough;
  final List<Ingredient> ingredients;
  final String tool;

  Pizza({
    required this.id,
    required this.name,
    required this.dough,
    required this.ingredients,
    required this.tool,
  });

  factory Pizza.fromJson(Map<String, dynamic> json) {
    return Pizza(
      id: json['id'] as int,
      name: json['name'] as String,
      dough: Dough.fromJson(json['dough'] as Map<String, dynamic>),
      ingredients: (json['ingredients'] as List)
          .map((i) => Ingredient.fromJson(i as Map<String, dynamic>))
          .toList(),
      tool: json['tool'] as String,
    );
  }
}

class Dough {
  final int id;
  final String name;
  final int? caloriesPerSlice;

  Dough({required this.id, required this.name, this.caloriesPerSlice});

  factory Dough.fromJson(Map<String, dynamic> json) {
    // Handle both "ID" (uppercase) and "id" (lowercase) for compatibility
    final idValue = json['ID'] ?? json['id'];
    return Dough(
      id: idValue as int,
      name: json['name'] as String,
      caloriesPerSlice: json['caloriesPerSlice'] as int?,
    );
  }
}

class Ingredient {
  final int id;
  final String name;
  final int? caloriesPerSlice;
  final bool? vegetarian;

  Ingredient({
    required this.id,
    required this.name,
    this.caloriesPerSlice,
    this.vegetarian,
  });

  factory Ingredient.fromJson(Map<String, dynamic> json) {
    // Handle both "ID" (uppercase) and "id" (lowercase) for compatibility
    final idValue = json['ID'] ?? json['id'];
    return Ingredient(
      id: idValue as int,
      name: json['name'] as String,
      caloriesPerSlice: json['caloriesPerSlice'] as int?,
      vegetarian: json['vegetarian'] as bool?,
    );
  }
}

class PizzaRecommendation {
  final Pizza pizza;
  final int? calories;
  final bool? vegetarian;

  PizzaRecommendation({required this.pizza, this.calories, this.vegetarian});

  factory PizzaRecommendation.fromJson(Map<String, dynamic> json) {
    return PizzaRecommendation(
      pizza: Pizza.fromJson(json['pizza'] as Map<String, dynamic>),
      calories: json['calories'] as int?,
      vegetarian: json['vegetarian'] as bool?,
    );
  }
}
